gateway/src/main/kotlin/com/gitlab/kordlib/gateway/Intent.kt

```diff
@@ -149,83 +152,97 @@ enum class Intent(val code: Int) {
  * A set of [intents][Intent] to be used while [identifying][Identify] a [Gateway] connection to communicate the events the client wishes to receive.
  */
 @Serializable(with = IntentsSerializer::class)
-data class Intents internal constructor(val code: Int) {
+data class Intents internal constructor(var code: Int = 0) {
 
     val intents = Intent.values().filter { code and it.code != 0 }.toSet()
 
+
     operator fun contains(intent: Intent) = intent in intents
 
     /**
      * Returns an [Intents] that added the [intent] to this [code].
      */
-    operator fun plus(intent: Intent): Intents = when {
-        code and intent.code == intent.code -> this
-        else -> Intents(this.code or intent.code)
-    }
+    operator fun plus(intent: Intent): Intents = Intents(this.code or intent.code)
+
+
+    /**
+     * Returns an [Intents] that added [code]s  in [Intents] to this [code].
+     */
+    operator fun plus(intent: Intents): Intents = Intents(this.code or intent.code)
+
+    /**
+     * Returns an [Intents] that removed the [intent] from this [code].
+     */
+    operator fun minus(intent: Intent): Intents = Intents(code xor intent.code)
```

Comment:
This is incorrect:

Intents(0B1100) - Intent(0B0010):
```
1100
0010
------- XOR
1110 <--expected--> 1100
```
If you want to inline this function into a single expression you should use `Intents XOR Intent AND Intents` instead:
```
1100
0010
------- XOR
1110 
1100 (Intents)
-------- AND
1100
```
use case where intent is in intents:
```
1100
0100
------- XOR
1000 
1100 (Intents)
-------- AND
1000
```